src/main/java/eu/hbp/mip/services/AlgorithmService.java

```diff
@@ -0,0 +1,284 @@
+        }
+
+        if (exaremeAlgorithms != null) {
+            algorithms.addAll(exaremeAlgorithms);
+            logger.LogUserAction("Loaded " + exaremeAlgorithms.size() + " exareme algorithms");
+        } else {
+            logger.LogUserAction("Fetching exareme algorithms failed");
+        }
+        if (mipengineAlgorithms != null) {
+            algorithms.addAll(mipengineAlgorithms);
+            logger.LogUserAction("Loaded " + mipengineAlgorithms.size() + " mipengine algorithms");
+        } else {
+            logger.LogUserAction("Fetching mipengine algorithms failed");
+        }
+        if (galaxyAlgorithms != null) {
+            algorithms.addAll(galaxyAlgorithms);
+            logger.LogUserAction("Loaded " + galaxyAlgorithms.size() + " galaxy algorithms");
+        } else {
+            logger.LogUserAction("Fetching galaxy workflows failed");
+        }
+
+        List<String> disabledAlgorithms = new ArrayList<>();
+        try {
+            disabledAlgorithms = getDisabledAlgorithms();
+        } catch (IOException e) {
+            logger.LogUserAction("The disabled algorithms could not be loaded.");
+        }
+
+        // Remove any disabled algorithm
+        ArrayList<ExaremeAlgorithmDTO> allowedAlgorithms = new ArrayList<>();
+        for (ExaremeAlgorithmDTO algorithm : algorithms) {
+            if (!disabledAlgorithms.contains(algorithm.getName())) {
+                allowedAlgorithms.add(algorithm);
+            }
+        }
+
+        int algorithmsRemoved = algorithms.size() - allowedAlgorithms.size();
+        if (algorithmsRemoved > 0){
+            logger.LogUserAction("Removed "+ (algorithmsRemoved) +" disabled algorithms");
+        }
+
+        this.algorithmDTOS = allowedAlgorithms;
+    }
+
+
+    @EnableAsync
+    public static class AlgorithmAggregator {
+
+        private final AlgorithmService algorithmService;
+
+        public AlgorithmAggregator(AlgorithmService algorithmService){
+            this.algorithmService = algorithmService;
+        }
+        @Async
+        @Scheduled(fixedDelayString = "${services.algorithmsUpdateInterval}000")
+        public void scheduleFixedRateTaskAsync() throws InterruptedException {
+            algorithmService.update(new Logger("AlgorithmAggregator","(GET) /algorithms"));
+        }
+    }
+
+    public  String getEngineName(String algorithmName){
+        Optional<ExaremeAlgorithmDTO> exaremeAlgorithmDTO  = this.algorithmDTOS.stream().filter(algorithmDTO -> algorithmDTO.getName().equals(algorithmName)).findAny();
+        if (exaremeAlgorithmDTO.isPresent()) return getEngineNameForSpecificAlgorithm(exaremeAlgorithmDTO.get());
+        else throw new BadRequestException("Algorithm: " + algorithmName + " does not exist.");
+    }
+
+    private String getEngineNameForSpecificAlgorithm(ExaremeAlgorithmDTO algorithmDTO){
```

Comment:
I think in Java you can have the same function name with a different parameter, so this method can be named `getAlgorithmEngineType` as well.